src/util/exception_utils.h

```diff
@@ -52,4 +52,13 @@ class system_exceptiont
   }
 };
 
+class deserialization_exceptiont
```

Comment:
See other PR: this should likely derive from some common invalid-input exception base class.